nano/node/noderpcconfig.cpp

```diff
@@ -0,0 +1,77 @@
+#include <nano/node/noderpcconfig.hpp>
+
+#include <nano/lib/blocks.hpp>
+#include <nano/lib/config.hpp>
+#include <nano/lib/jsonconfig.hpp>
+#include <nano/lib/rpcconfig.hpp>
+
+nano::error nano::node_rpc_config::serialize_json (nano::jsonconfig & json) const
+{
+	json.put ("version", json_version ());
+	json.put ("enable_sign_hash", enable_sign_hash);
+	json.put ("max_work_generate_difficulty", nano::to_string_hex (max_work_generate_difficulty));
+	json.put ("rpc_path", rpc_path);
+	json.put ("rpc_in_process", rpc_in_process);
+	return json.get_error ();
+}
+
+nano::error nano::node_rpc_config::deserialize_json (bool & upgraded_a, nano::jsonconfig & json, boost::filesystem::path const & data_path)
+{
+	auto version_l (json.get_optional<unsigned> ("version"));
+	if (!version_l)
+	{
+		version_l = 1;
+		json.put ("version", *version_l);
+		json.erase ("frontier_request_limit");
+		json.erase ("chain_request_limit");
+
+		// Don't migrate enable_sign_hash as this is not needed by the external rpc process, but first save it.
+		json.get_optional ("enable_sign_hash", enable_sign_hash, false);
+
+		json.erase ("enable_sign_hash");
+		json.erase ("max_work_generate_difficulty");
+
+		migrate (json, data_path);
+
+		json.erase ("io_threads");
+
+		json.put ("enable_sign_hash", enable_sign_hash);
+		json.put ("max_work_generate_difficulty", nano::to_string_hex (max_work_generate_difficulty));
+		json.put ("rpc_path", get_default_rpc_filepath ());
+		bool rpc_in_process_l;
+		json.get_optional<bool> ("rpc_in_process", rpc_in_process_l);
+		if (!rpc_in_process)
+		{
+			json.put ("rpc_in_process", true);
+		}
+
+		upgraded_a = true;
+	}
+
+	json.get_optional<bool> ("enable_sign_hash", enable_sign_hash);
+	std::string max_work_generate_difficulty_text;
+	json.get_optional<std::string> ("max_work_generate_difficulty", max_work_generate_difficulty_text);
+	if (!max_work_generate_difficulty_text.empty ())
+	{
+		nano::from_string_hex (max_work_generate_difficulty_text, max_work_generate_difficulty);
+	}
+	json.get_optional<std::string> ("rpc_path", rpc_path);
+	json.get_optional<bool> ("rpc_in_process", rpc_in_process);
+	return json.get_error ();
+}
+
+void nano::node_rpc_config::migrate (nano::jsonconfig & json, boost::filesystem::path const & data_path)
+{
```

Comment:
Should `migrate` remove unused json properties, like "port", from the old config? 

By the way, I had a non-standard RPC port number in config.json, but the new rpc_config.json contains the default 7076 after the upgrade.

Reply:
Yes it should! Thanks, done.

I wonder if you still had "version" : "1" in the old rpc config (I have made that mistake). We have done an upgrade to it this release which I am hijacking, so it won't upgrade correctly if the version is present.

Reply:
I removed the version field as part of the test.

Reply:
hmm and the rpc_config.json file was deleted beforehand? The port seems to migrate correctly for me, perhaps you could try again after I push all the changes.

Reply:
yeah, will do, i'll try migrating from a proper v18 installation, seems like a more reliable way to test